src/miner.cpp

```diff
@@ -559,14 +563,48 @@ void static BitcoinMiner(CWallet *pwallet)
                     std::lock_guard<std::mutex> lock{m_cs};
                     return cancelSolver;
                 };
-                try {
-                    // If we find a valid block, we rebuild
-                    if (EhOptimisedSolve(n, k, curr_state, validBlock, cancelled))
-                        break;
-                } catch (EhSolverCancelledException&) {
-                    LogPrint("pow", "Equihash solver cancelled\n");
-                    std::lock_guard<std::mutex> lock{m_cs};
-                    cancelSolver = false;
+
+                if (solver == "tromp") {
+                    // Create solver and initialize it.
+                    equi eq(1);
```

Comment:
Currently `-genproclimit` is set up such that it will start multiple independent solvers. Now that there's a solver impl that supports internal threading, we should think about how to leverage that in a user-friendly way. Does not block.